### PR TITLE
ffmpeg: Add 5.1.3 library (with avfliter included)

### PR DESCRIPTION
**Pull request by lat9nq**
Required for yuzu-emu/yuzu#10283

This also creates a copy of the ffmpeg_compile.sh script. It was already in the FFmpeg packages, but now it is in plain sight (and now is subject to Git tracking as well).

Script saw some changes: shallow clones the FFmpeg repo, and now finds the library versions for the requirements text file.

### CODE REVIEW

File: ffmpeg/ffmpeg_compile.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+fi
+
+cd FFmpeg
+
+AVCODEC_VER=$(grep '#define LIBAVCODEC_VERSION_MAJOR' libavcodec/version_major.h | sed 's/.* //g')
+AVUTIL_VER=$(grep '#define LIBAVUTIL_VERSION_MAJOR' libavutil/version.h | sed 's/.* //g')
+SWSCALE_VER=$(grep '#define LIBSWSCALE_VERSION_MAJOR' libswscale/version_major.h | sed 's/.* //g')
+AVFILTER_VER=$(grep '#define LIBAVFILTER_VERSION_MAJOR' libavfilter/version_major.h | sed 's/.* //g')
+
+REQUIRED_DLLS="avcodec-${AVCODEC_VER}.dll;avutil-${AVUTIL_VER}.dll;libwinpthread-1.dll;swscale-${SWSCALE_VER}.dll;avfliter-${AVFILTER_VER}.dll"
```
Comment (danilaml) [inline ffmpeg/ffmpeg_compile.sh:25]: Typo: avf**li**ter -> avf**il**ter

**Comment by danilaml** [comment]
This build also requires `libiconv-2.dll`. Probably because environment changes and now iconv is detected as present on a build machine but is not copies in the script like pthreads is. Since it wasn't needed before, I assume just adding `--disable-iconv` and rebuilding should be enough. Perhaps `-disable-autodetect` altogether, although the current build might depend on some implicitly enabled libs, so this needs some testing/experiments.

**Comment by danilaml** [comment]
Also, for some reason this build doesn't include any filters besides (a)buffer(sink) (these are added manually, IIRC), which is strange. Can you check your `libavfilter/filter_list.c`? They should all be listed there (at least `yadif`). Try adding `--enable-filter=yadif` to be 100% sure it gets there.

**Comment by lat9nq** [comment]
@danilaml Sorry it took me a few days to notice the comments. Check if the most recent version fixes these issues.

**Comment by danilaml** [comment]
@lat9nq Thanks. The updated build appears to be working fine!

**Comment by danilaml** [comment]
@lat9nq I think it's OK to merge it now.

**Comment by Squall-Leonhart** [comment]
This PR has likely broken Pokemon Brilliant Diamond and Shiny Pearl's video decoding, as reported [here](https://github.com/yuzu-emu/yuzu/issues/10571)

**Pull request merged**